components/admin_console/blockable_link/blockable_link.tsx

```diff
@@ -2,34 +2,31 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import PropTypes from 'prop-types';
 import {NavLink} from 'react-router-dom';
 
 import {browserHistory} from 'utils/browser_history';
 
-export default class BlockableLink extends React.Component {
-    static propTypes = {
+type Props = {
 
-        /*
-         * Bool whether navigation is blocked
-         */
-        blocked: PropTypes.bool.isRequired,
-
-        /*
-         * String Link destination
-         */
-        to: PropTypes.string.isRequired,
+    /*
+   * Bool whether navigation is blocked
+   */
+    blocked: boolean;
 
-        actions: PropTypes.shape({
+    /*
+   * String Link destination
+   */
+    to: string;
+    actions: {
 
-            /*
-             * Function for deferring navigation while blocked
-             */
-            deferNavigation: PropTypes.func.isRequired,
```

Comment:
is it going to use [this](https://github.com/mattermost/mattermost-webapp/blob/master/actions/admin_actions.jsx#L273)? because that is taking a parameter and returning an object